apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/dao/state/publickey/PublicKeyTableProducer.java

```diff
@@ -98,4 +101,35 @@ private void init() {
             return genesisPublicKeyTable;
         }
     }
+
+    public void onRescanBegan(@Observes @BlockEvent(BlockEventType.RESCAN_BEGIN) Block block) {
```

Comment:
I guess, that this code is not in a proper place (PublicTableProducer is not related to rescan/hotswap weld events)